ml-agents/mlagents/trainers/trainer_controller.py

```diff
@@ -71,8 +73,31 @@ def __init__(self, env_path, run_id, save_freq, curriculum_folder,
                 docker_target_name=docker_target_name,
                 run_id=run_id)
             if env_path is not None:
-                env_path = '/{docker_target_name}/{env_name}'.format(
-                    docker_target_name=docker_target_name, env_name=env_path)
+                """
+                Comments for future maintenance:
+                    Some OS/VM instances (e.g. COS GCP Image) mount filesystems 
+                    with COS flag which prevents execution of the Unity scene, 
+                    to get around this, we will copy the executable into the 
+                    container.
+                """
+                # Navigate in docker path and find env_path and copy it.
+                for f in glob.glob('/{docker_target_name}/*'.format(
```

Comment:
Also do you always need to do this copying? Or should you first check if you are on a instance that has "noexec" flag, and if true you do this copying.

Reply:
It  is not harmful to do the copying. I think checking within the code would make the code a bit uglier.

Reply:
Make sense.